draft-ietf-quic-transport.md

```diff
@@ -317,23 +317,23 @@ lower-numbered streams of the same type in the same direction.
 
 QUIC allows for an arbitrary number of streams to operate concurrently.  An
 endpoint limits the number of concurrently active incoming streams by limiting
-the maximum stream ID (see {{stream-limit-increment}}).
+the number of streams (see {{stream-limit-increment}}).
 
-The maximum stream ID is specific to each endpoint and applies only to the peer
-that receives the setting. That is, clients specify the maximum stream ID the
-server can initiate, and servers specify the maximum stream ID the client can
-initiate.  Each endpoint may respond on streams initiated by the other peer,
-regardless of whether it is permitted to initiate new streams.
+The stream limit is specific to each endpoint and applies only to the peer that
+receives the setting. That is, clients limit the number of streams the server
```

Comment:
Hm I'm not in love with the "number of streams" terminology. With "maximum stream ID" it was very obvious what was being limited. If the max stream ID was X, you clearly could not open a steam with ID X+1 (or X+4 but whatever :>).

With a limit on the "number of a streams" this is less clear. If a server limits the client to 10, presumably the idea is that this is equivalent to a max stream id limit of the 10th client initiated stream ID. But it also seems like you could read this limit to permit the client to open any 10 steams. Like 0, 40, 80, 120, ... I think this is not the intent, but the terminology makes this much less clear to me.

For example, the text below:

"A peer MUST NOT initiate a stream with a higher stream ID than the largest limit
maximum stream ID it has received. "

Clearly the limit applies to stream ID, but using "number of streams" seems to obscure that.

Reply:
I agree that this isn't great, but it's an artifact of encoding stream type in the ID.  A decision I really don't want to revisit, BTW.

Given that the number of streams can be very easily determined from the ID with `(id+3)/4`, or the test for whether an stream can be opened is `id/4 >= limit`, I think that this is quite workable.

I've tweaked the text you point at.  Hopefully this is clearer.